indexing-service/src/main/java/io/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -164,12 +165,20 @@ public Firehose connect(InputRowParser inputRowParser) throws IOException, Parse
       List<String> dims;
       if (dimensions != null) {
         dims = dimensions;
+      } else if (inputRowParser.getParseSpec().getDimensionsSpec().hasCustomDimensions()) {
+        dims = inputRowParser.getParseSpec().getDimensionsSpec().getDimensions();
```

Comment:
Kafka firehose factories seem to be the only ones that account for exclusions prior to this patch. Their dim exclusions also exclude "feed", sorry for my unfamiliarity with the firehose process, but is that a Kafka specific thing?